addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -820,7 +838,13 @@ var SnippetEditor = Widget.extend({
             firstOptions.forEach(option => {
                 $optionsSectionBtnGroup.prepend(option.$el);
             });
-            $optionsSection.toggleClass('d-none', options.length === 0);
+            let visibleOption = false;
+            let i = 0;
+            while (!visibleOption && i < options.length) {
+                visibleOption = await options[i]._computeVisibility();
+                i++;
+            }
+            $optionsSection.toggleClass('d-none', (options.length === 0 || !visibleOption));
```

Comment:
**BEFORE MERGE** at least remove this diff (we can add the right version later)

Mmh I am not sure this is a good idea. First this does not consider `_computeWidgetVisibility` and mainly... this is done only once the options are instantiated for the first time if I am not mistaken? This should be done when options visibility is (re)computed. And there should be comments around this, please :)

Reply:
I removed the diff but I did not come with a better solution yet. I don't really know where to add a TODO for this but the empty "Page Options" present on every page might be a good reminder.

I'll try to look into it once I've resolved the other comments if that's Ok